app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorFragment.kt

```diff
@@ -63,8 +65,23 @@ class CreditCardEditorFragment : SecureFragment(R.layout.fragment_credit_card_ed
         creditCardEditorView.bind(creditCardEditorState)
     }
 
+    /**
+     * Close any open dialogs or menus and reauthenticate if the fragment is paused and
+     * the user is not navigating to [CreditCardsManagementFragment].
+     */
+    override fun onPause() {
+        menu.close()
+
+        redirectToCreditCardReAuth(
```

Comment:
I don't quite understand this. I don't think we need to reauth if we were adding a new credit card. Perhaps if we were editing, but I am really not sure about this experience. I would follow up on this with some product or UX feedback.

Reply:
This is the same experience as logins. It wasn't clear to me that we were securing this differently than logins

Reply:
Update: I have confirmed that we are supposed to be doing this the same as logins.